server/routerlicious/packages/lambdas/src/utils/documentLambdaCheckpointManager.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IQueuedMessage } from "@fluidframework/server-services-core";
+import { ICheckpoint } from "./checkpointHelper";
+
+export interface IDocumentCheckpointManager {
+	checkpointInfo: ICheckpoint;
+	noActiveClients: boolean;
+	useCheckpointIdleTimer: boolean;
+
+	updateCheckpointMessages(message: IQueuedMessage);
+	clearCheckpointIdleTimer(): void;
+	resetCheckpointTimer(): void;
+	updateCheckpointIdleTimer(
+		timeout: number,
+		idleTimeCheckpoint: (message: IQueuedMessage) => void,
+		isDocumentCorrupt?: boolean,
+	): void;
+}
+
+export class DocumentCheckpointManager implements IDocumentCheckpointManager {
+	checkpointInfo: ICheckpoint = {
```

Comment:
Maybe this is personal opinion, but I think from a `class` design standpoint, it would probably be better to hide some of these internal details behind methods instead of exposing them for public read/write.

For example, instead of `this.documentCheckpointManager.checkpointInfo.rawMessagesSinceCheckpoint++` it would maybe be better to expose a method like `this.documentCheckpointManager.incrementRawMessageCounter()`

Reply:
I see that in some cases, `checkpointInfo` can contain lots more optional details. Perhaps having the `incrementRawMessageCounter()` as well as a getter would suffice?

```ts
private checkpointInfo: ICheckpointInfo = {...};
//...
public incrementRawMessageCounter(): number {
  return checkpointInfo.rawMessagesSinceCheckpoint++;
}
public get checkpointInfo(): ICheckpointInfo {
  return this.checkpointInfo;
}
```

Reply:
Good point. I agree. Just added some updates to add more access modifiers/methods